pyscriptjs/src/components/pyconfig.ts

```diff
@@ -27,25 +41,65 @@ export class PyConfig extends BaseEvalElement {
         super();
     }
 
-    connectedCallback() {
+    extractFromSrc() {
+        return validateConfig(readTextFromPath(this.getAttribute('src')));
+    }
+
+    extractFromInline() {
         this.code = this.innerHTML;
         this.innerHTML = '';
+        return validateConfig(this.code);
+    }
+
+    getDefaultConfig()
+    {
+        return inJest() ? defaultConfig : JSON.parse(defaultConfig);
+    }
+
+    connectedCallback() {
+        let loadedValues: object = {};
+        let srcConfig: object = {};
+        let inlineConfig: object = {};
 
-        const loadedValues = jsyaml.load(this.code);
-        if (loadedValues === undefined) {
-            this.values = {
-                autoclose_loader: true,
-                runtimes: [DEFAULT_RUNTIME_CONFIG]
-            };
-        } else {
-            // eslint-disable-next-line
-            // @ts-ignore
-            this.values = loadedValues;
+        // load config from src and inline
+        if (this.hasAttribute('src') && this.innerHTML!=='')
+        {
+            srcConfig = this.extractFromSrc();
+            inlineConfig = this.extractFromInline();
+            // first make config from src whole if it is partial
+            srcConfig = mergeConfig(srcConfig, this.getDefaultConfig());
+            // then merge inline config and config from src
+            loadedValues = mergeConfig(inlineConfig, srcConfig);
+            logger.info('config set from src attribute and inline both, merging', JSON.stringify(loadedValues));
+        }
+        // load config from src only
+        else if (this.hasAttribute('src'))
+        {
+            srcConfig = this.extractFromSrc();
+            logger.info('config set from src attribute, merging with default', srcConfig);
+            loadedValues = mergeConfig(srcConfig, this.getDefaultConfig());
+        }
+        // load config from inline
+        else if (this.innerHTML!=='')
+        {
+            inlineConfig = this.extractFromInline();
+            logger.info('config set from inline, merging with default', inlineConfig);
+            loadedValues = mergeConfig(inlineConfig, this.getDefaultConfig());
```

Comment:
What are the conditions adding here? Feels like we could avoid all these conditions and just have functions return empty configs if they are not defined. So all the code above is reduced to:

```
srcConfig = this.extractFromSrc();
inlineConfig = this.extractFromInline();
// first make config from src whole if it is partial
srcConfig = mergeConfig(srcConfig, this.getDefaultConfig());
// then merge inline config and config from src
loadedValues = mergeConfig(inlineConfig, srcConfig);
```

Reply:
yeah, this is a good suggestion, changing